scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/FileOperations.java

```diff
@@ -64,6 +90,35 @@ protected FileOperations(Compression compression, String mimeType) {
     final ReadableFile readableFile = toReadableFile(resourceId);
 
     final Reader<V> reader = createReader();
+
+    Long bytes = diskBufferBytes.get();
+    if (bytes != null && bytes > 0) {
+      final long fileSize = readableFile.getMetadata().sizeBytes();
+      final long prevSize = diskBufferBytes.getAndUpdate(prev -> prev > 0 ? prev - fileSize : prev);
+
+      // Buffer available, an update was made
+      if (prevSize > 0) {
+        LOG.info("Buffering SMB source file {}, size = {}B", resourceId, fileSize);
```

Comment:
is this log going to be really noisy for the millions-of-files use case? maybe the counter is enough of a signal?

Reply:
Yeah probably redundant after I added counters. Will change to `debug`.